consoles/video_stream.pm

```diff
@@ -0,0 +1,353 @@
+# Copyright 2021 Marek Marczykowski-Górecki
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package consoles::video_stream;
+
+use Mojo::Base -strict, -signatures;
+
+use base 'consoles::network_console';
+
+use List::Util 'max';
+use Time::HiRes qw(usleep);
+
+use Try::Tiny;
+use testapi 'get_var';
+use bmwqemu;
+
+# speed limit: 30 keys per second
+use constant STREAM_TYPING_LIMIT_DEFAULT => 30;
+
+use constant DV_TIMINGS_CHECK_INTERVAL => 3;
+
+use constant STALL_THRESHOLD => 4;
+
+my $CHARMAP = {
+    "\t" => 'tab',
+    "\n" => 'ret',
+    "\b" => 'backspace',
+    "\e" => 'esc',
+    " " => 'space',
+};
+
+sub screen ($self, @) {
+    return $self;
+}
+
+sub disable_video ($self) {
+    my $ret = 0;
+    if ($self->{ffmpeg}) {
+        kill(TERM => $self->{ffmpegpid});
+        close($self->{ffmpeg});
+        $self->{ffmpeg} = undef;
+        $ret = waitpid($self->{ffmpegpid}, 0);
+        $self->{ffmpegpid} = undef;
+    }
+    return $ret;
+}
+
+sub disable ($self, @) {
+    my $ret = $self->disable_video;
+    if ($self->{input_pipe}) {
+        close($self->{input_pipe});
+        waitpid($self->{inputpid}, 0);
+    }
+    return $ret;
+}
+
+sub _v4l2_ctl ($device, $cmd) {
+    my @cmd = ("v4l2-ctl", "-d", $device, "-k");
```

Comment:
Maybe long options are a little clearer? At least I had to lookup "device" and  "koncise". Well, no strong opinion
```suggestion
    my @cmd = ("v4l2-ctl", "--device", $device, "--concise");
```